File: android/src/main/kotlin/com/crefter/yuchengplugin/yucheng_ble/YuchengSportDataConverter.kt
package com.crefter.yuchengplugin.yucheng_ble

import YuchengSportData
import com.google.gson.Gson

private class YuchengSportBean(
    val sportStartTime: Long,    // start timestamp (seconds)
val  sportEndTime: Long,  // end timestamp (seconds)
    val sportStep: Int,  // number of steps (steps)
    val sportDistance: Int, // distance (meters)
    val sportCalorie: Int, // calories (kcal)
)

class YuchengSportDataConverter(private val gson: Gson) {
    fun convert(sportDataBean: Any?): YuchengSportData {
        val converted = gson.fromJson(sportDataBean.toString(), YuchengSportBean::class.java)
        return YuchengSportData(
            startTimeStamp = converted.sportStartTime,
            endTimeStamp = converted.sportEndTime,
            distance = converted.sportDistance.toLong(),
            steps = converted.sportStep.toLong(),
            calories = converted.sportCalorie.toLong(),
        )
    }
}

File: ios/Classes/YuchengSportDataConverter.swift
//
//  YuchengSportDataConverter.swift
//  Pods
//
//  Created by Maxim Zarechnev on 18.08.2025.
//

import YCProductSDK

final class YuchengSportDataConverter: Sendable {
    func convert(sportDataFromDevice: YCHealthDataStep) -> YuchengSportData {
        let sportData = YuchengSportData(startTimeStamp: Int64(sportDataFromDevice.startTimeStamp * 1000), endTimeStamp: Int64(sportDataFromDevice.endTimeStamp * 1000), distance: Int64(sportDataFromDevice.distance), steps: Int64(sportDataFromDevice.step), calories: Int64(sportDataFromDevice.calories))
        return sportData
    }
}
